### ISSUE TITLE
Refactor dungeon generation.

### ISSUE THREAD

**Issue by ndouglas** [ndouglas/azymus#46]
Currently, I have three "algorithms" for creating a dungeon: "simple", which is a slightly modified version of the code from the tutorial, "empty", which just creates an absolutely minimal map, and "random", which ostensibly chooses a random generator (but just points to the "simple" generator right now.

The game is currently based around the "simple" generator, which causes some problems (e.g. #17).

Ultimately, I would like:

- a number of different generators that would generate different kinds of maps:
  - wilderness (several different kinds, for different biomes)
  - dungeon (probably several of these, just for variety)
  - maze (possibly a couple different kinds)
  - cave systems
  - town/city
  - combinations (cave/town? wilderness/maze?)
  - other ?
- some way of tying possible denizens to a given map, based on what would be somewhat compatible/reasonable to find there
- some way of influencing the inhabitants through the environment, a sort of environmental determinism
- etc

So, basically, I need to generate the map before populating it, then populate it from the bottom up and run a few turns before setting the player loose.

**Comment by ndouglas** [comment]
One possibility is to dump a bunch of critters in and have them evolve over a bunch of generations.  Is that way too front-heavy?  Too much startup delay?  IDK.  I do think it means I'd have to generate dungeons that are far more than 80x25, 80x50, or even 160x100 to be interesting... but I've had that thought anyway.  Maybe I need to generate _systems_ of interconnected maps.  Definitely not Dwarf Fortress levels of crazy, but something larger than a single map at a time.
